### PR TITLE
React query integration

### PR DESCRIPTION
**Pull request by ErezBiren**
* Add a lot of new API requests
* Add HealthStatus component with Mocked data
image

* get Values from API
image

@chad11111

### CODE REVIEW

File: dashboard/src/components/InstalledPackages/InstalledPackageCard.tsx [hunk 1]
```diff
@@ -31,21 +31,26 @@ export default function InstalledPackageCard({
     setIsMouseOver(false);
   };
 
+  const handleOnClick = () => {
+
+    console.log(release)
```
Comment (ImmanuelSegol) [inline dashboard/src/components/InstalledPackages/InstalledPackageCard.tsx:36]: remove

File: dashboard/src/components/revision/RevisionDetails.tsx [hunk 2]
```diff
@@ -98,6 +100,8 @@ function RevisionDetails({ release }: RevisionDetailsProps) {
     console.error("unInstallConfirmed not implemented"); //todo: implement
   };
 
+  console.log("111" + JSON.stringify(release));
```
Comment (ImmanuelSegol) [inline dashboard/src/components/revision/RevisionDetails.tsx:103]: remove

File: dashboard/src/components/revision/RevisionDetails.tsx [hunk 3]
```diff
@@ -85,9 +87,9 @@ function RevisionDetails({ release }: RevisionDetailsProps) {
     setIsOpenUninstallModal(true);
   };
 
-  const rollback = ()=>{
-    console.error("not implemented")
-  }
+  const rollback = () => {
+    console.error("not implemented");
```
Comment (ImmanuelSegol) [inline dashboard/src/components/revision/RevisionDetails.tsx:91]: throw an error no?

**Comment by ErezBiren** [comment]
@chad11111

File: dashboard/src/components/InstalledPackages/InstalledPackageCard.tsx [hunk 4]
```diff
@@ -31,21 +32,25 @@ export default function InstalledPackageCard({
     setIsMouseOver(false);
   };
 
+  const handleOnClick = () => {
+    console.log(release);
```
Comment (chad11111) [inline dashboard/src/components/InstalledPackages/InstalledPackageCard.tsx:36]: remove plz

File: dashboard/src/components/revision/RevisionsList.tsx [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { getAge } from "../../timeUtils";
 import StatusLabel from "../common/StatusLabel";
 
 type RevisionsListProps = {
-  releaseRevisions: ReleaseRevision[];
+  releaseRevisions: ReleaseRevision[] | undefined;
```
Comment (chad11111) [inline dashboard/src/components/revision/RevisionsList.tsx:7]: just use ?

**Review by chad11111 — CHANGES_REQUESTED** [review]

File: dashboard/src/components/revision/RevisionsList.tsx (hunk 5 above)
Comment (ErezBiren) [reply dashboard/src/components/revision/RevisionsList.tsx]: done

File: dashboard/src/components/InstalledPackages/InstalledPackageCard.tsx (hunk 4 above)
Comment (ErezBiren) [reply dashboard/src/components/InstalledPackages/InstalledPackageCard.tsx]: done

**Pull request merged**